src/marten/routing/map.cr

```diff
@@ -20,13 +20,25 @@ module Marten
 
       # Inserts a new path into the routes map.
       #
-      # The target associated with the considered path must be either a handler (subclass of `Marten::Handlers::Base`)
-      # or another `Marten::Routing::Map` instance (in case of nested routes maps). Each <path, target> pair must be
-      # given a name that will be used to uniquely identify the route.
-      def path(path : String, target : Marten::Handlers::Base.class | Map, name : String | Symbol) : Nil
-        if name.empty?
-          raise Errors::InvalidRuleName.new("Route names cannot be empty")
-        end
+      # The target associated with the considered path must be a `Marten::Routing::Map`
+      # instance. The <path, target> pair has an optional
+      # name that will be prepended to each <path, target> pair inside the `Marten::Routing::Map`.
+      def path(path : String, target : Map, name : String | Symbol = "") : Nil
```

Comment:
I think we could signify that the `name` is optional by making the argument and the corresponding instance variable nillable. That way we could keep the non-empty check for map paths.